msupply-foundation/open-msupply
# 3189 add lines by item

Fixes #3189

# 👩🏻‍💻 What does this PR do? 
 

Implements the add by item/edit line version of the inbound return edit modal, and updates the table cells to be inputs:
![Screenshot 2024-03-15 at 2 36 23 PM](https://github.com/msupply-foundation/open-msupply/assets/55115239/7cbddafe-b74d-4f97-8eef-5fc5571b5818)


# 🧪 How has/should this change been tested? 


- Add a new inbound return
- From the detail view, click Add Item
- Select an item
- [ ] A first draft line should appear
- [ ] Clicking `Add batch`, or using the `+` key, should add more lines
- [ ] You can input the line info
- Ok & Next, input reason/comment, OK
- [ ] The line is added to the return
- [ ] Clicking the line re-opens the modal
- [ ] You can update fields for that line
- [ ] Setting return quantity to 0 deletes the line
